pkg/cmd/server/api/helpers.go

```diff
@@ -17,6 +17,85 @@ import (
 	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 )
 
+var (
+	openShiftFeatureNames map[string]string
+)
+
+func init() {
+	openShiftFeatureNames = make(map[string]string, len(KnownOpenShiftFeatures))
+	openShiftFeatureNames["builder"] = "Builder"
+	openShiftFeatureNames["s2i"] = "S2I Builder"
+	openShiftFeatureNames["console"] = "Web Console"
+}
+
+// OpenShiftFeatureName returns name pleasant to read for given feature.
+func OpenShiftFeatureName(feature string) string {
+	lowerCased := strings.ToLower(feature)
+	if name, ok := openShiftFeatureNames[lowerCased]; ok {
+		return name
+	}
+	return feature
+}
+
+// Add extends feature list with given valid items. They are appended
+// unless already present.
+func (fl *FeatureList) Add(items ...string) error {
+	var (
+		err error
+		fs  = util.NewStringSet(*fl...)
+	)
+	for _, item := range items {
+		lowerCased := strings.ToLower(item)
+		if fs.Has(item) {
+			continue
+		}
+		if _, exists := openShiftFeatureNames[item]; !exists {
+			err = fmt.Errorf("unknown openshift feature %q", item)
```

Comment:
This is the actual check. If the item doesn't belong to keys of the map, it's invalid.